### PR TITLE
feat: support multi app keys

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [x] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: lib/api/client.js [hunk 1]
```diff
@@ -204,37 +213,161 @@ module.exports = class Client {
...
+
   /**
    * 获取 jsapi_ticket, 并在有效期内自动缓存
    *  - get_jsapi_ticket
    * @return {String} jsapiTicket
    * @see https://open-doc.dingtalk.com/doc2/detail.htm?treeId=172&articleId=104966&docType=1
    */
   * getJSApiTicket() {
-    let jsapiTicket = yield this.callFn(this.cache.get, [ 'jsapiTicket' ]);
+    let jsapiTicket = yield this.callFn(this.cache.get, [ 'jsapiTicket' ], this.cache);
```
Comment (fengmk2) [inline lib/api/client.js]: 这是 bugfix，修复丢失 this 的问题

File: lib/api/client.js [hunk 2]
```diff
@@ -204,37 +213,161 @@ module.exports = class Client {
...
    */
-  * getAccessToken() {
-    let accessToken = yield this.callFn(this.cache.get, [ 'accessToken' ]);
+  * getAccessToken(options) {
+    options = options || {};
+    const appid = options.appid;
+    const appsecret = options.appsecret;
+    const corpid = options.corpid || this.options.corpid;
+    const corpsecret = options.corpsecret || this.options.corpsecret;
+    const cacheKey = 'accessToken_' + utility.md5(appid ? `${appid}|${appsecret}` : `${corpid}|${corpsecret}`);
```
Comment (fengmk2) [inline lib/api/client.js]: 主要是改进多 app 之后，cache key 的问题。

**Comment by fengmk2** [comment]
ci 挂了是老的问题了

**Comment by atian25** [comment]
先合了吧，API 跑 CI 不稳定，APPID 容易失效

**Comment by atian25** [comment]
1.1.0

**Pull request merged**